pilot/pkg/bootstrap/server.go

```diff
@@ -883,7 +883,19 @@ func (s *Server) initRegistryEventHandlers() {
 	s.ServiceController().AppendServiceHandler(serviceHandler)
 
 	if s.configController != nil {
-		configHandler := func(_ config.Config, curr config.Config, event model.Event) {
+		configHandler := func(prev config.Config, curr config.Config, event model.Event) {
+			defer func() {
+				if event != model.EventDelete {
+					s.statusReporter.AddInProgressResource(curr)
+				} else {
+					s.statusReporter.DeleteInProgressResource(curr)
+				}
+			}()
+			// For update events, trigger push only if spec has changed.
+			if event == model.EventUpdate && !needsPush(prev, curr) {
```

Comment:
Trying to look for places where metadata may matter:

https://github.com/istio/istio/blob/83b692a1ea44dca0f27c745f4c2970fc4d847332/pilot/pkg/config/kube/crdclient/client.go#L343 - I think not relevant since this is *before* the push trigger

https://github.com/istio/istio/blob/7d4a1366f5ce1ca8eef67ccd05975f4c02850e8b/pilot/pkg/config/kube/gateway/conversion.go#L93 - This is a real usage of labels. We need to make label changes trigger push for TLSRoute, TCPRoute, HTTPRoute. Can we also put a comment on L93 that if we add support for others (IE virtual service) we need to update it maybe?

Name/Namespace change - not possible

AuthzPolicy dry run: https://github.com/istio/istio/blob/f393a2dde0910ed71749b1604b6e093caa723547/pilot/pkg/security/authz/builder/builder.go#L158 - this needs a change

Possibly workload entry annotations matter? not certain: https://github.com/istio/istio/blob/358a5da1a43146502c5b563c7870d3d8d2f18ada/pilot/pkg/controller/workloadentry/workloadentry_controller.go#L250

Gateway annotation: https://github.com/istio/istio/blob/77366e7a2b077d426df043669f09fe23d4a22c7e/pilot/pkg/model/push_context.go#L1740

Reply:
I will take a look